index.js

```diff
@@ -3,26 +3,38 @@ const path = require('path');
 
 const pluginName = 'Chunks2JsonWebpackPlugin';
 
+const defaultOptions = {
+    // ignore files emitted by HMR by default
+    excludeFile: /\.hot-update\.js$/,
+    // group chunks by extension
+    chunkGroupName: filename => /\.([a-z0-9]+(\.map)?)(\?.*)?$/.exec(filename)[1],
+    outputDir: process.cwd(),
+    filename: 'build-manifest.json',
+    // generate contents to save to manifest file
+    objectToString: result => JSON.stringify(result),
+    publicPath: ''
+};
+
 class Chunks2JsonWebpackPlugin {
     constructor(options) {
-        this.options = options;
+        this.options = Object.assign({}, defaultOptions, options);
         this.result = {};
     }
     apply(compiler) {
         compiler.hooks.emit.tap(pluginName, compilation => {
-            compilation.chunks.forEach((chunk) => {
-                if (this.result[chunk.name] === undefined) {
+            this.result = {};
+            compilation.chunks.forEach(chunk => {
+                if (!this.result[chunk.name]) {
                     this.result[chunk.name] = {};
                 }
-                chunk.files.forEach((filename) => {
-                    if (filename.endsWith('css')) {
-                        this.result[chunk.name].css = `/${filename}`;
-                    } else if (filename.endsWith('js')) {
-                        this.result[chunk.name].js = `/${filename}`;
-                    } else if (filename.endsWith('js.map')) {
-                        this.result[chunk.name].jsMap = `/${filename}`;
-                    } else if (filename.endsWith('css.map')) {
-                        this.result[chunk.name].cssMap = `/${filename}`;
+                chunk.files.forEach(filename => {
+                    const exclude = typeof this.options.excludeFile === 'function' ?
+                        this.options.excludeFile(filename, chunk) :
+                        this.options.excludeFile.test(filename);
+                    if (!exclude) {
+                        const ext = this.options.chunkGroupName(filename, chunk);
+                        if (!this.result[chunk.name][ext]) this.result[chunk.name][ext] = [];
+                        this.result[chunk.name][ext].push(this.options.publicPath + filename);
```

Comment:
Please use ES6 string interpolation here - 

```javascript
this.result[chunk.name][ext].push(`${this.options.publicPath}${filename}`);
```

Reply:
Why do you think it's better? My point of view is that one should use template strings when there is some static text around variables interpolation. This case is too simple.

Reply:
Because when reading it, you see right away, that this should be string. 
But see the point that there's not static text. But even then, I prefer consistency when it comes to string concatenation.